### DISCUSSION TITLE
php bin/console PHP Fatal error:

### DISCUSSION THREAD

**Discussion by StephaneCursan** [symfony/symfony#49658] [Q&A]
Hello everyone,

I regained control of an "old" website (developed with Symfony 3.4). I had restored my Mac (now under macOS Monterey 12.6). And I recreated a Symfony environment (composer create-project symfony/framework-standard-edition psf_gironde) and pasted the original project into the new htdocs. Everything seems to work fine except the blog page. There is a problem with the database access. I got several errors like "Unknown Database...", "Access denied for user..." etc.

![image](https://user-images.githubusercontent.com/46920921/224304806-0770eff6-3fd4-48c0-bdd5-d7c22d323ae5.png)
[_One of the errors displayed when accessing the blog page. Browser screenshot_]

I understood that the database was not created. But when I run a "php bin/console doctrine:database:create" I only get a list of errors as long like a day without bread. But in fact I think it's the php bin/console command that is causing me trouble. No matter what I try ("cache:clear", "about", "server:run"), the result is always this famous long list of errors.

![image](https://user-images.githubusercontent.com/46920921/224305031-e0f0a20d-2b96-40cd-a0bf-7c69876d4912.png)
[_Extract from the "php bin/console" errors list. Screenshot of the terminal._]

The version of php, according to the terminal (php -v) is 8.1.3. The one selected on cpanel is 7.4. My long research on the Web leads me to believe that there could be a compatibility problem with the php version. But I don't know how to change it using the command line.

Otherwise I had found an idea here by installing php-xml (apt install php-xml). I ran the command but I get the error "The operation could not be completed. Unable to locate a Java Runtime." And there too I have no idea :/

Could someone please help me? I am not an expert and I am under pressure. Thanks a lot

**Comment by ghost** [comment]
Was the app you "regained control of" running on 7.4? If so then install 7.4. 

> But I don't know how to change it using the command line.

I am a docker user for a long time but as far as I remember you can install multiple php versions on you host and when running PHP scripts just do not type `php -S localhost:8080` but the full path to the specific php version executable e.g. `/usr/bin/php7.4 -S localhost:8080`. If you use Symfony Server CLI then I'm sure they also support multiple PHP versions.

**Comment by lyrixx** [comment]
I suggest you to use the same version between HTTP & Cli and also between prod & dev.

If you cannot, you can use the composer option called [platform](https://getcomposer.org/doc/06-config.md#platform) to simulate a PHP version 

---

Then run `composer update`, and the error should disappear
